@W-22570447 - Enabling ANC for shopper agent. (#3854)
SalesforceCommerceCloud/pwa-kit: packages/template-retail-react-app/CHANGELOG.md

> ## v10.1.0-dev

Reviewer: Duplicate `## v10.1.0-dev` heading. The new bullet was added under its own section above the existing one (line 4 still has the same heading), so the file now opens with two `## v10.1.0-dev` blocks back-to-back. Merge into the existing section instead.

```suggestion
## v10.1.0-dev
- [Feature] Enable customer context for Shopper Agent: pass SLAS access + refresh tokens to Core via a same-origin Token Bridge proxy on conversation start, and reset the embedded messaging session on guest ↔ registered transitions so the agent never picks up a stale shopper identity.
```

-   [Feature] Enable customer context for Shopper Agent: pass SLAS access + refresh tokens to Core via a same-origin Token Bridge proxy on conversation start, and reset the embedded messaging session on guest ↔ registered transitions so the agent never picks up a stale shopper identity.

> 

Reviewer: Duplicate `## v10.1.0-dev` heading. The new bullet was added under its own section above the existing one (line 4 still has the same heading), so the file now opens with two `## v10.1.0-dev` blocks back-to-back. Merge into the existing section instead.

```suggestion
## v10.1.0-dev
- [Feature] Enable customer context for Shopper Agent: pass SLAS access + refresh tokens to Core via a same-origin Token Bridge proxy on conversation start, and reset the embedded messaging session on guest ↔ registered transitions so the agent never picks up a stale shopper identity.
```

Author: merged

## v10.1.0-dev

> - [Feature] Enable customer context for Shopper Agent: pass SLAS access + refresh tokens to Core via a same-origin Token Bridge proxy on conversation start, and reset the embedded messaging session on guest ↔ registered transitions so the agent never picks up a stale shopper identity.

Reviewer: [NIT] The changelog says the Token Bridge passes "SLAS access + refresh tokens to Core", but after the HttpOnly rewrite the client only ever sends the access token in the body (and only in non-HttpOnly mode). Refresh token is read server-side from `cc-nx_{siteId}` / `cc-nx-g_{siteId}` cookies in `token-bridge.js:196-206`. Worth rewording so a future reader doesn't chase a client-side refresh-token path that no longer exists.

```suggestion
- [Feature] Enable customer context for Shopper Agent: on conversation start, call Core's Token Bridge via a same-origin PWA Kit proxy that forwards the shopper's SLAS session (access token in the body for non-HttpOnly mode; both tokens read server-side from `cc-at_*` / `cc-nx_*` cookies in HttpOnly mode) and reset the embedded messaging session on guest ↔ registered transitions so the agent never picks up a stale shopper identity.
```
